### DISCUSSION TITLE
Official list of highlight capture names

### DISCUSSION THREAD

**Discussion by leonerd** [tree-sitter/tree-sitter#2025] [Q&A] [answered]
I see in `tree-sitter/config.json` there's a list of some highlight capture names, and also I can find many examples of `queries/highlight.scm` files to take the names from. But is there an overall list of all that are used, with descriptions of what they mean?

Sometimes it can be hard to guess quite what different languages really mean for these things. For example, words like "attribute", "property", "module" might have different meanings in different places. It would be good to have one central place where these ideas are explained out, so as to get a bit more consistency between them.

**Comment by ahelwer** [comment, marked as answer]
The list of capture names supported by the `tree-sitter highlight` command can be found here: https://github.com/tree-sitter/tree-sitter/blob/2949b210b9c6b930a8f00e35b3d8db281769df68/cli/src/highlight.rs#L150-L171

In general the list of capture names is client-specific, though. nvim-treesitter has a different list. Other than GitHub itself I don't know of clients that use this built-in list of names.

**Reply by leonerd** [reply]
> In general the list of capture names is client-specific, though. nvim-treesitter has a different list.

Oh, I see. That's a little unfortunate. I was hoping to just write a single `highlights.scm` file and have it generally useful to any client. If they use different names that does make that task somewhat harder.

Any useful advice on how to write my `highlights.scm` file anyway, to be most useful?

**Reply by ahelwer** [reply]
It'll be specific to your language. Just experiment and try to find something that looks nice under a few different colorschemes.
